test/test_document_store.py

```diff
@@ -265,29 +265,34 @@ def test_extended_filter(document_store_with_docs):
         "name": ["filename5", "filename3"],
     }
     documents_simplified_filter = document_store_with_docs.get_all_documents(filters=filters_simplified)
-    assert documents == documents_simplified_filter
+    # Order of returned documents might differ
+    assert len(documents) == len(documents_simplified_filter) and all(
+        doc in documents_simplified_filter for doc in documents
+    )
 
     filters = {
         "$and": {
             "date_field": {"$lte": "2020-12-31", "$gte": "2019-01-01"},
-            "$or": {"name": {"$in": ["filename5", "filename3"]}, "numeric_field": {"$lte": 5}},
+            "$or": {"name": {"$in": ["filename5", "filename3"]}, "numeric_field": {"$lte": 5.0}},
         }
     }
     documents = document_store_with_docs.get_all_documents(filters=filters)
     assert len(documents) == 2
     filters_simplified = {
         "date_field": {"$lte": "2020-12-31", "$gte": "2019-01-01"},
-        "$or": {"name": ["filename5", "filename3"], "numeric_field": {"$lte": 5}},
+        "$or": {"name": ["filename5", "filename3"], "numeric_field": {"$lte": 5.0}},
     }
     documents_simplified_filter = document_store_with_docs.get_all_documents(filters=filters_simplified)
-    assert documents == documents_simplified_filter
+    assert len(documents) == len(documents_simplified_filter) and all(
+        doc in documents_simplified_filter for doc in documents
+    )
 
     filters = {
         "$and": {
             "date_field": {"$lte": "2020-12-31", "$gte": "2019-01-01"},
             "$or": {
                 "name": {"$in": ["filename5", "filename3"]},
-                "$and": {"numeric_field": {"$lte": 5}, "$not": {"meta_field": {"$eq": "test2"}}},
+                "$and": {"numeric_field": {"$lte": 5.0}, "$not": {"meta_field": {"$eq": "test2"}}},
```

Comment:
This seems to be one of the only two tests where NOT is used with weaviate. I think, we should add at least one additional test to check that the inversion of conditions works as expected. That should address nested conditions within NOT to check that AND, OR and NOT are correctly inverted and as part of that, I would like to see how > is converted to <=.

Reply:
Added a test case with multiple all logical operations (AND, OR, NOT) embedded inside a not operation.